lib/asyncapiSchemaFormatParser.js

```diff
@@ -0,0 +1,46 @@
+const Ajv = require('ajv');
+const ParserError = require('./errors/parser-error');
+const asyncapi = require('@asyncapi/specs');
+const { improveAjvErrors } = require('./utils');
+
+module.exports = {
+  parse,
+  getMimeTypes
+};
+
+async function parse({ message, oryginalAsyncAPIDocument, fileFormat, parsedAsyncAPIDocument }) {
+  const ajv = new Ajv({
+    jsonPointers: true,
+    allErrors: true,
+    schemaId: 'id',
+    logger: false,
+  });
+  const payloadSchema = preparePayloadSchema(asyncapi[parsedAsyncAPIDocument.asyncapi]);
+  const validate = ajv.compile(payloadSchema);
+  const valid = validate(message.payload);
+
+  if (!valid) throw new ParserError({
+    type: 'validation-errors',
```

Comment:
I'd be more specific here. Something like:

```suggestion
    type: 'schema-validation-errors',
```